annotation/tests/test_tasks_services.py

```diff
@@ -0,0 +1,321 @@
+import re
+from unittest.mock import Mock, call, patch
+
+import pytest
+from fastapi import HTTPException
+
+from annotation.errors import FieldConstraintError
+from annotation.jobs.services import ValidationSchema
+from annotation.models import File, ManualAnnotationTask
+from annotation.tasks.services import (
+    check_cross_annotating_pages,
+    validate_files_info,
+    validate_task_info,
+    validate_user_actions,
+    validate_users_info,
+)
+
+
+@pytest.mark.parametrize(
+    ("validation_type", "is_validation"),
+    (
+        (ValidationSchema.validation_only, True),
+        ("other_type", False),
+        ("other_type", True),
+    ),
+)
+def test_validate_task_info(
+    validation_type: ValidationSchema,
+    is_validation: bool,
+):
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session, patch(
+        "annotation.tasks.services.validate_users_info"
+    ) as mock_validate_users_info, patch(
+        "annotation.tasks.services.validate_files_info"
+    ) as mock_validate_files_info:
+
+        db_session = mock_session()
+        task_info = {"is_validation": is_validation}
+
+        validate_task_info(db_session, task_info, validation_type)
+
+        mock_validate_users_info.assert_called_once_with(
+            db_session, task_info, validation_type
+        )
+        mock_validate_files_info.assert_called_once_with(db_session, task_info)
+
+
+def test_validate_task_info_invalid_task_info():
+    with patch("sqlalchemy.orm.Session", spec=True) as mock_session:
+        db_session = mock_session()
+        task_info = {"is_validation": False}
+        validation_type = ValidationSchema.validation_only
+
+        with pytest.raises(FieldConstraintError):
+            validate_task_info(db_session, task_info, validation_type)
+
+
+@pytest.mark.parametrize(
+    ("is_validation"),
+    (
+        (True),
+        (False),
+    ),
+)
```

Comment:
```suggestion
@pytest.mark.parametrize("is_validation", (True, False))
```

Reply:
fixed;)

Reply:
still the same

Reply:
I usually fix the issue locally first, then I write 'ok,' 'fixed,' or 'done' once it's resolved before pushing the changes. Sorry about that!